### ISSUE TITLE
Header as pane

### ISSUE THREAD

**Issue by allanabiud** [mierak/rmpc#190]
### Description

I've seen what you've done with the lyrics and implemented it as a pane. 
It would be great if the same could be done with the header to allow it be positioned at the bottom together with the progress bar and keeping its row customisations of course.

### Proposed solution

Implement the header as a pane similar to queue, album art and lyrics.

**Comment by mierak** [comment]
I definitely want to do something about making them configurable. I dont have specific implementation in mind yet, but am unsure whether making them into panes is the way.

**Comment by abiud254** [comment]
I also wanted to ask about the `browser_song_format`. In the documentation it's stated it takes the same format as `song_table_format`. Does this mean it can have columns as well and you can use properties on it as well, cause I tried and it didn't work.

I also tried things like `album_art_position` and `album_art_width_percent`, and they didn't work.

Also, styling `(kind: Property(Status(Bitrate)))` like so `(kind: Property(Status(Bitrate)), style: (fg: "red", modifiers: "Bold")),` didn't work.

**Comment by mierak** [comment]
> I also tried altering album_art_position and album_art_width_percent, and nothing changed.

Those two are deprecated in favor of the tabs config which does the same and more. My bad on not pruning them properly from the docs, will fix that.

> I also wanted to ask about the browser_song_format. In the documentation it's stated it takes the same format as song_table_format.

No, it just means that the it takes the same formatters as the `song_table_format` except with styles being ignored, but I see how its misleading, will also fix that. You can check the (default theme)[https://mierak.github.io/rmpc/release/reference/theme/] at the bottom to see what its supposed to look like.

> Also, styling (kind: Property(Status(Bitrate))) like so (kind: Property(Status(Bitrate)), style: (fg: "red", modifiers: "Bold")), didn't apply any sort of styling.

That is a genuine bug, will fix that in a bit

Thanks for the feedback!

**Comment by mierak** [comment]
The styles not working fixed in https://github.com/mierak/rmpc/pull/192. All mentions of `album_art_position` and `album_art_width_percent` were removed too.

**Comment by abiud254** [comment]
> No, it just means that the it takes the same formatters as the `song_table_format` except with styles being ignored, but I see how its misleading, will also fix that.

I guess I was confused as to what counts as styles and formatters. I was just going over the documentation for song_table_format and thought those options were available to browser_song_format as well.

I just wish it was clearer which ones work for browser_song_format.

**Comment by mierak** [comment]
I took the everything as pane approach here https://github.com/mierak/rmpc/pull/208. As it says in the PR, it lets you move everything around by specifying `layout` in your theme. Its not completely ready yet, but would be great to get a second pair of eyes on it while its still baking, since it was quite a big change and I might have broken some stuff.

**Comment by mierak** [comment]
The layout is configured in the theme config. The tabs work alongside the layout and are put into a special `TabContent` pane in the layout, you can see that in the example in the PR.

**Comment by abiud254** [comment]
So I don't need the tabs configuration in the base config anymore?

**Comment by abiud254** [comment]
I was able to get it working.

**Comment by abiud254** [comment]
![rmpc-layout](https://github.com/user-attachments/assets/24ec141d-5cf8-4269-bfc3-7583332c984d)
Works just fine from my testing.

Was just confused on how this will conflict with the tab configurations.
Also on the sizes. Weird that percentages work alongside what I would assume are pixels. It being standardized would be great.
And also be able to have a border on the header maybe.

**Comment by mierak** [comment]
Thanks! Glad to see its working somewhat and sorry for the confusion.

Sizes are either with a percent sign, ie `"50%"` or in terminal cells (rows or columns) and these are written just as a number inside quotes ie. `"5"`. Thats from https://github.com/mierak/rmpc/pull/203 feature and is [documented here](https://mierak.github.io/rmpc/next/configuration/tabs/#panes). Sadly theres not much more I can do to make it better other than documenting it better which I will try to do.

Borders are already planned after this is merged!

**Comment by abiud254** [comment]
> Thanks! Glad to see its working somewhat and sorry for the confusion.

> Sizes are either with a percent sign, ie "50%" or in terminal cells (rows or columns) and these are written just as a number inside quotes ie. "5". Thats from https://github.com/mierak/rmpc/pull/203 feature and is [documented here](https://mierak.github.io/rmpc/next/configuration/tabs/#panes). This is inteded so you can for example make lyrics take only 5 rows and some other pane the rest of the space. Sadly theres not much more I can do to make it better other than documenting it better which I will try to do.

Yeah I was confused that they can both be used together. I expected wither percentage sizes or terminal cells.  But better documentation will make it more understandable.

> Borders are already planned after this is merged!

Nice.

**Comment by mierak** [comment]
https://github.com/mierak/rmpc/pull/208 is merged, I dont notice any issues. If you have some please open a new issue. The feature is documented here https://mierak.github.io/rmpc/next/configuration/layout/.

**Comment by mierak** [comment]
Sorry for necroing this again, https://github.com/mierak/rmpc/pull/239 in progress. It supports advanced borders definition, if you want I would appreciate some testing/feedback. Theres still some work (mainly docs I suppose) before merging, but the PR includes examples and description.
